Assets/Scripts/Entity/EntityHealth.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Mirror;
+using UnityEngine.Events;
+
+public class EntityHealth : NetworkBehaviour
+{
+    [SyncVar]
+    private float health = 30f;
+    public float Health => health;
+
+    public UnityEvent OnDie;
+    public UnityEvent OnTakeDamage;
+
+    #region Server
+
+    [Server]
+    public virtual void TakeDamage(float damage)
+    {
+        OnTakeDamage.Invoke();
+        Debug.Log(gameObject.name + " took " + damage + " damage");
+        health -= damage;
+        Debug.Log("Health: " + health);
+        if (health <= 0)
```

Comment:
This shouldn't be or should rarely be a problem here (taking an extra frame of damage doesn't really matter), but make sure you don't run into any floating-point precision errors:

```
[Server]
public void TakeDamage(float damage)
{
    // ...

    if (Mathf.Approximately(health, 0f)) { return; }

    health = Mathf.Max(health - damage, 0f);

    if (Mathf.Approximately(health, 0f)) { 
        Die();
    }
}
```